web-local/src/lib/svelte-query/invalidation.ts

```diff
@@ -33,21 +33,51 @@ export const invalidateAfterReconcile = async (
             getNameFromFile(affectedPath)
           )
         ),
+        getInvalidationsForPath(queryClient, affectedPath),
       ])
       .flat()
   );
 };
 
+const getInvalidationsForPath = (
+  queryClient: QueryClient,
+  filePath: string
+) => {
+  const name = getNameFromFile(filePath);
+  if (filePath.startsWith("/dashboards")) {
+    return invalidateMetricsViewData(queryClient, name);
+  } else {
+    return invalidateProfilingQueries(queryClient, name);
+  }
+};
+
 export const invalidateMetricsViewData = (
   queryClient: QueryClient,
-  instanceId: string,
   metricsViewName: string
 ) => {
   return queryClient.refetchQueries({
     predicate: (query) =>
       typeof query.queryKey[0] === "string" &&
       query.queryKey[0].startsWith(
-        `/v1/instances/${instanceId}/metrics-views/${metricsViewName}/`
+        `/v1/instances/[a-zA-Z0-9-]+/metrics-views/${metricsViewName}/`
       ),
   });
 };
+
+export function invalidationForProfileQueries(queryHash, name: string) {
+  const r = new RegExp(
+    `/v1/instances/[a-zA-Z0-9-]+/queries/[a-zA-Z0-9-]+/tables/${name}`
```

Comment:
This actually doesn't appear to invalidate the timeseries queries in profiling, which has the following query key:

```
[
    "/v1/instances/default/queries/timeseries",
    {
        "tableName": "rill_product_events_model",
        "timestampColumnName": "__time",
        "pixels": 92
    }
]
```

Would it be possible to add a check for this kind of key here?

Reply:
Fixed by making timeseries match the other requests.